pkg/hope/hypervisors/proxmox_hypervisor.go

```diff
@@ -0,0 +1,275 @@
+package hypervisors
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"slices"
+	"strings"
+	"time"
+)
+
+import (
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+import (
+	"github.com/Eagerod/hope/pkg/hope"
+	"github.com/Eagerod/hope/pkg/packer"
+	"github.com/Eagerod/hope/pkg/proxmox"
+)
+
+type ProxmoxHypervisor struct {
+	node hope.Node
+
+	pc *proxmox.ApiClient
+}
+
+func (p *ProxmoxHypervisor) Initialize(node hope.Node) error {
+	p.node = node
+
+	errs := []error{}
+	useInsecureTransport := false
+	pm := ParameterMap(node.Parameters)
+	if insecure, ok := pm["INSECURE"]; ok {
```

Comment:
So this whole thing is basically the exact same as esxi. Sarting to feel like room/need for some more complete tooling here.